app/src/main/java/com/nextcloud/talk/translate/TranslateActivity.kt

```diff
@@ -0,0 +1,276 @@
+import com.nextcloud.talk.models.json.translations.TranslationsOverall
+import com.nextcloud.talk.users.UserManager
+import com.nextcloud.talk.utils.ApiUtils
+import com.nextcloud.talk.utils.bundle.BundleKeys
+import com.nextcloud.talk.utils.database.user.CapabilitiesUtilNew
+import io.reactivex.Observer
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import org.json.JSONArray
+import java.util.Locale
+import javax.inject.Inject
+
+@AutoInjector(NextcloudTalkApplication::class)
+class TranslateActivity : BaseActivity() {
+    private lateinit var binding: ActivityTranslateBinding
+
+    @Inject
+    lateinit var ncApi: NcApi
+
+    @Inject
+    lateinit var userManager: UserManager
+
+    var fromLanguages = arrayOf<String>()
+
+    var toLanguages = arrayOf<String>()
+
+    var text: String? = null
+
+    var check: Int = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        NextcloudTalkApplication.sharedApplication!!.componentApplication.inject(this)
+        binding = ActivityTranslateBinding.inflate(layoutInflater)
+
+        setupActionBar()
+        setContentView(binding.root)
+        setupSystemColors()
+        setupTextViews()
+        setupSpinners()
+        getLanguageOptions()
+        translate(null, Locale.getDefault().language)
+    }
+
+    private fun setupActionBar() {
+        setSupportActionBar(binding.translationToolbar)
+        binding.translationToolbar.setNavigationOnClickListener {
+            onBackPressed()
+        }
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setDisplayShowHomeEnabled(true)
+        supportActionBar?.setIcon(ColorDrawable(resources!!.getColor(R.color.transparent)))
+        supportActionBar?.title = resources!!.getString(R.string.translation)
+        viewThemeUtils.material.themeToolbar(binding.translationToolbar)
+    }
+
+    private fun setupTextViews() {
+        val original = binding.originalMessageTextview
+        val translation = binding.translatedMessageTextview
+
+        viewThemeUtils.talk.themeIncomingMessageBubble(original, grouped = true, deleted = false)
+        viewThemeUtils.talk.themeIncomingMessageBubble(translation, grouped = true, deleted = false)
+
+        original.movementMethod = ScrollingMovementMethod()
+        translation.movementMethod = ScrollingMovementMethod()
+
+        val bundle = intent.extras
+        binding.originalMessageTextview.text = bundle?.getString(BundleKeys.KEY_TRANSLATE_MESSAGE)
+        text = bundle?.getString(BundleKeys.KEY_TRANSLATE_MESSAGE)
+    }
+
+    private fun getLanguageOptions() {
+        val currentUser: User = userManager.currentUser.blockingGet()
+        val json = JSONArray(CapabilitiesUtilNew.getLanguages(currentUser).toString())
+        Log.i(TAG, "json is: $json")
```

Comment:
log can be removed

Reply:
fixed via https://github.com/nextcloud/talk-android/pull/3025/commits/b779a01c274919c464a40ec3e71cfc947071bb92